src/infrastructure/jira/jira-service.ts

```diff
@@ -154,23 +187,128 @@ class JiraService {
 				]),
 				connectInstallation,
 			);
-		} catch (error) {
-			if (error instanceof JiraClientNotFoundError) {
-				const newAttachedDesignUrlIssuePropertyValue: AttachedDesignUrlV2IssuePropertyValue =
-					{
-						url,
-						name: displayName,
-					};
-				const value = JSON.stringify([newAttachedDesignUrlIssuePropertyValue]);
-				await jiraClient.setIssueProperty(
+		}
+	};
+
+	deleteDesignUrlInIssueProperties = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		await Promise.all([
+			await this.deleteAttachedDesignUrlInIssuePropertiesIfPresent(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+			await this.deleteFromAttachedDesignUrlV2IssueProperties(
+				issueIdOrKey,
+				design,
+				connectInstallation,
+			),
+		]);
+	};
+
+	/**
+	 * @internal
+	 * Only visible for testing. Please use {@link deleteDesignUrlInIssueProperties}
+	 */
+	deleteAttachedDesignUrlInIssuePropertiesIfPresent = async (
+		issueIdOrKey: string,
+		design: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		try {
+			const response = await jiraClient.getIssueProperty(
+				issueIdOrKey,
+				propertyKeys.ATTACHED_DESIGN_URL,
+				connectInstallation,
+			);
+
+			const storedUrl = ensureString(response.value);
+
+			if (storedUrl === design.url) {
+				await jiraClient.deleteIssueProperty(
 					issueIdOrKey,
-					propertyKeys.ATTACHED_DESIGN_URL_V2,
-					value,
+					propertyKeys.ATTACHED_DESIGN_URL,
 					connectInstallation,
 				);
-			} else {
-				throw error;
 			}
+		} catch (error) {
+			if (error instanceof JiraClientNotFoundError) {
+				return; // Swallow not found errors
+			}
+			throw error;
+		}
+	};
+
+	/**
+	 * @internal
+	 * Only visible for testing. Please use {@link saveDesignUrlInIssueProperties}
+	 */
+	deleteFromAttachedDesignUrlV2IssueProperties = async (
+		issueIdOrKey: string,
+		{ url, displayName }: AtlassianDesign,
+		connectInstallation: ConnectInstallation,
+	): Promise<void> => {
+		let response: GetIssuePropertyResponse;
+		try {
+			response = await jiraClient.getIssueProperty(
+				issueIdOrKey,
+				propertyKeys.ATTACHED_DESIGN_URL_V2,
+				connectInstallation,
+			);
+		} catch (error) {
+			if (error instanceof JiraClientNotFoundError) {
+				return; // Swallow not found errors
+			}
+			throw error;
+		}
+		const storedAttachedDesignUrlIssuePropertyValues = JSON.parse(
```

Comment:
**nitpick:** Currently, the code does _unsafe_ type assertion and then validation. It would be safer to do validation and then type assertion.

Interestingly, we could use [a type guard with an assertion signature](https://learntypescript.dev/07/l7-assertion-signature) to solve this problem (which was an interesting finding for me).

Please, consider the following:

```ts
deleteFromAttachedDesignUrlV2IssueProperties = async (
  issueIdOrKey: string,
  { url, displayName }: AtlassianDesign,
  connectInstallation: ConnectInstallation,
): Promise<void> => {
  // ...
  const storedAttachedDesignUrlIssuePropertyValues = JSON.parse(response.value);
  assertAttachedDesignUrlV2IssuePropertyValueType(storedAttachedDesignUrlIssuePropertyValues);

  // Type should be inferred by the TS compiler here thanks to the type guard with an assertion signature.
  // ...
}

private assertAttachedDesignUrlV2IssuePropertyValueType = (
  value: unknown,
) assertys value is AttachedDesignUrlV2IssuePropertyValue[] => {
  // ...
};
```

**PS:** Please, also pay attention that you do not `ensureString` since `JSON.parse` throws if any non-string input is provided.

Reply:
Updated, thanks for the suggestion @akostevich-atlassian